### DISCUSSION TITLE
NextJS Trailing Slash

### DISCUSSION THREAD

**Discussion by hect1c** [nextauthjs/next-auth#2847] [Help]
Hey,

In my `next.config.js` I have `trailingSlash: true` so this forces any non slash url to redirect to a trailing slash. I noticed all of my next-auth requests essentially always got through a 308 redirect due to this trailing slash. Is there way for me to have next-auth adhere to this next config by default to avoid the 308 redirects? This would save a lot of unnecessary requests that is processed for each next-auth api request.

**Comment by balazsorban44** [comment]
I think it would require us to read your `next.config.js` file and do runtime checks, which sounds like an unnecessary overhead for everyone not interested in this 😕

Even your problem wouldn't go away, the solution would just be tucked away in `next-auth`.

**Reply by hect1c** [reply]
Yes that would be unnecessary overhead, could it maybe just be an extra configuration option that we could pass in NextAuthConfigOptions ?

Ideally I would like to just have next-auth know to use the trailingSlash as to avoid redirects and hit the correct url endpoint directly. Albeit I wouldn't say this is very critical as the redirects are fairly quick. So more of a nice to have, but I'm sure your time is better spent on getting v4 completed. 

Thanks for the quick responses as always and keep up the great work.

**Reply by balazsorban44** [reply]
thanks! I cannot promise anything, but this might be the right place to implement such option

https://github.com/nextauthjs/next-auth/blob/beta/src/server/lib/extend-res.ts

or you could see if https://next-auth.js.org/configuration/callbacks#redirect-callback is anything helpful

**Reply by Oxicode** [reply]
anything news?

**Reply by rexwebmedia** [reply]
any updates on this ?

![image](https://github.com/user-attachments/assets/6be03b88-9cc5-4359-a332-e1f0c586ca63)

**Reply by Chimochikhin** [reply]
Any News?
